compiler/IREmitter/IREmitterHelpers.h

```diff
@@ -58,6 +58,31 @@ class Intrinsics {
     };
 };
 
+struct KnownFunction {
+    enum class Type {
+        // Symbols available for direct comparison
+        Symbol,
+
+        // Symbols that are computed by calling another function, and then cached
+        CachedSymbol,
+    };
+
+    Type type;
+    std::string name;
+
+    // Implicit constructor from a string for backwards compability
+    KnownFunction(std::string name) : KnownFunction(Type::Symbol, std::move(name)) {}
+
+    KnownFunction(Type type, std::string name) : type{type}, name{std::move(name)} {}
```

Comment:
WDYT about making this function private?

Reply:
Works for me!